eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/AbstractImmutableMap.java

```diff
@@ -519,4 +523,44 @@ public ImmutableShortBag collectShort(ShortFunction<? super V> shortFunction)
     {
         return this.countByEach(function, Bags.mutable.empty()).toImmutable();
     }
+
+    public static class ImmutableKeySet<K> extends AbstractImmutableSet<K>
```

Comment:
@johnjstuart this class seems unused (and also useless, why wrap an `ImmutableSet` to make it Immutable?

Reply:
@johnjstuart yes I realised later that you use it in other classes and not here. But see my comments below about that usage, I don't think it is needed.